src/controllers/commands_controller.js

```diff
@@ -114,28 +114,48 @@ CommandsController.post('/', function (request, response) {
           .then(function(thingBody) {
             let jsonBody = JSON.parse(thingBody);
             let match = payload.param.toUpperCase();
+            let thingfound = false; 
             for (var i = 0; i < jsonBody.length; i++) {
               var obj = jsonBody[i];
               let name = obj.name.toUpperCase();
               if (name == match) {
-                payload.href = obj.properties.on.href;
+                thingfound = true;
                 break;
               }
             }
-            if (payload.href != '') {
+            if (thingfound) {
               // Returning 201 to signify that the command was mapped to an
```

Comment:
nit: can you move this comment to where it returns 201?

Reply:
fixed